.github/workflows/check-odo.yml

```diff
@@ -5,41 +5,131 @@ on:
+
+          # Update cache
+          echo "$lastCommitId" >> ../lastCommit
+          popd
+      - name: Check New Changes
+        id: cache-last-commit
+        uses: actions/cache@v4
+        with:
+          path: lastCommit
+          key: lastCommit-${{ hashFiles('lastCommit') }}
+
+  should-update-odo:
+    runs-on: ubuntu-latest
+    needs: should-check-odo
+    if: ${{ needs.should-check-odo.outputs.repo-cache-hit != 'true' || github.event_name != 'schedule' }}
+    outputs:
+      odo-version-changed: ${{ (steps.check-odo-version.outputs.odo-version-changed) }}
+      last-release-tag: ${{ needs.should-check-odo.outputs.repo-last-release-tag }}
+      last-release-version: ${{ steps.check-odo-version.outputs.last-release-version }}
+      nightly-build-version: ${{ steps.check-odo-version.outputs.nightly-build-version }}
+      nightly-build-commit-id: ${{ needs.should-check-odo.outputs.repo-last-commit-id }}
     steps:
       - name: Check Out Code
         uses: actions/checkout@v4
-      - name: Get latest ODO version
+      - name: Check if ODO should be updated
+        id: check-odo-version
         run: |
-          echo "REPO_ODO_VERSION=$(cat src/tools.json | jq -r .odo.version)" >> $GITHUB_ENV
-          LATEST_TOOL_RELEASE_RESP=$(gh release --repo ${{ env.TOOL_REPO }} view --json tagName,url)
-          echo "LATEST_TOOL_RELEASE=$(echo ${LATEST_TOOL_RELEASE_RESP} | jq -r .tagName | sed 's|v||')" >> $GITHUB_ENV
-          echo "LATEST_TOOL_URL=$(echo ${LATEST_TOOL_RELEASE_RESP} | jq -r .url)" >> $GITHUB_ENV
-      - name: Find existing PR for ODO version
-        run: |
-          echo PR_EXISTS=$(gh pr --repo ${{ github.repository }} list --state all --search "update odo ${{env.LATEST_TOOL_RELEASE}} in:title" --json url | jq length) >> $GITHUB_ENV
+          # Last Release Version
+          lastReleaseVersion="$(echo "${{needs.should-check-odo.outputs.repo-last-release-tag}}" | sed 's|v||')"
+          echo "last-release-version=$lastReleaseVersion" >> "$GITHUB_OUTPUT"
+
+          # Currently used ODO version
+          currentOdoVersion="$(cat src/tools.json | jq -r .odo.version)"
+          echo "current-version=$currentOdoVersion" >> "$GITHUB_OUTPUT"
+
+          # Construct Nightly Build Version
+          nightlyBuildVersion="$(echo "$lastReleaseVersion" | { IFS=. read b s p; echo "$b.$s.$((p+1))"; })-${{ needs.should-check-odo.outputs.repo-last-commit-build-number }}-nightly"
+          echo "nightly-build-version=$nightlyBuildVersion" >> "$GITHUB_OUTPUT"
+
+          if [[ "$currentOdoVersion" != "$nightlyBuildVersion" ]]; then
+            echo "odo-version-changed=true" >> "$GITHUB_OUTPUT"
+          fi
+
+  update-odo:
+    runs-on: ubuntu-latest
+    needs: should-update-odo
+    if: ${{ needs.should-update-odo.outputs.odo-version-changed == 'true' }}
+    steps:
+      - name: Check Out Code
+        uses: actions/checkout@v4
+      - name: Check if the update PR already exists
```

Comment:
I got this error when running the workflow after already creating a PR :

`a pull request for branch "bump-odo-3.15.1-116-nightly" into branch "main" already exists:`

So I think there's something wrong

Reply:
Check please, you probably already have  this PR created (open or closed), no? (if this is not the first time you start the GH action)

Reply:
This is outdated now.